### ISSUE TITLE
Theme colors - Insufficient contrasts - light mode

### ISSUE THREAD

**Issue by luce-carevic** [elk-zone/elk#1365]
<img width="309" alt="" src="https://user-images.githubusercontent.com/6073336/213863517-1b05287b-becb-4fe2-b859-05beaa6e6835.png">

The theme colors provided in light mode are not contrasted enough with the white background. This is an accessibility issue.

The contrast ratio is around 3:1 for each color when at least 4,5:1 is expected by WCAG SC 1.4.3. Contrast (Minimum).

[Read Understanding SC 1.4.3](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).

The best solution would be to add and option for users to choose their own color (with a colorpicker).

**Comment by boehs** [comment]
cc @antfu (introduced theme colours in #1195

**Comment by magicalskye** [comment]
is this going to be solved? i would really like to use elk-zone because the mastodon web interface is a pain and all of the other web clients lack other features that i need. this is the most promising one so far but the colour options make it unusable for me. i am an obligate light mode user, and none of the options makes it possible for me to actually read the text of hashtags or links, because there is simply not enough contrast. my eyes just present this as a blurry mess :(

**Comment by luce-carevic** [comment]
It seems that @antfu you don't think adding a color picker would be a good idea (cf. [your comment in this issue](https://github.com/elk-zone/elk/pull/1195#issuecomment-1383778735)) ?

Another solution is to change the theme colors provided in the light theme to match the required contrast ratio but I would argue that the color picker is a better idea to suit everyone's need.

Dark mode is not suitable for everyone.

**Comment by simevidas** [comment]
Is it just me, or would it be better if the Sign in button switched to white text?

On the left is how it looks right now in a desktop browser. The color contrast is 4.88. On the right is how it would look if we switched the text color to white. The color contrast would be 3.22, which is insufficient (< 4.5). However, it looks *more* readable to me. The WCAG contrast algorithm has a known issue with light text on dark backgrounds, so maybe we shouldn’t look at the number but actually ask people what they think is better.

<img width="651" alt="Screenshot 2023-10-25 at 07 27 07" src="https://github.com/elk-zone/elk/assets/716405/b3118d4a-babe-4317-bb01-57ac3b3d6ccb">

**Comment by shuuji3** [comment]
I've also sometimes felt less readable on the left color combination. 

There is the other specification called ACPA, which mitigates the WCAG issue. So I tested the ACPA color contract checker and found out the right one actually has a better score.

- Black text: https://contrast.tools/?text=000&background=cc7e23&tab=apca
- White text: https://contrast.tools/?text=fff&background=cc7e23&tab=apca

I made a PR to change the button color to white #2449. We can also check the change in the live preview: https://deploy-preview-2449--elk-zone.netlify.app/.

**Comment by luce-carevic** [comment]
APCA is not fully tested, it is not a specification equivalent of WCAG. 

A number of disabled people have expressed that some colours that passed APCA (and failed the WCAG algorithm) are not readable to them. It should not be used instead of the WCAG algorithm

There is no perfect algorithm that will fix readability for everyone. That’s not possible since the needs vary. 

The best thing to do is to give users the choice of their theme colour and that’s what I’ve been suggesting in my issue.

**Comment by shuuji3** [comment]
@luce-carevic Sorry for the late reply. Thanks for sharing the insight for WCAG/APCA. My intention for the button text color change was only partial improvement for just a bit better contract to meet @simevidas's suggestion (maybe I should have created a different issue separated from this issue).

Yes, I agree with you that my PR is not enough solution to resolve this issue entirely, and to cover the needs of everyone, we need to provide a separate solution such as a color selector option or something like that.
